pybop/costs/base_cost.py

```diff
@@ -61,75 +63,105 @@ def __init__(self, problem: Optional[BaseProblem] = None):
 
     def __call__(
         self,
-        inputs: Union[Inputs, list],
+        inputs: Union[Inputs, list, np.ndarray],
         calculate_grad: bool = False,
         apply_transform: bool = False,
         for_optimiser: bool = False,
-    ) -> Union[float, tuple[float, np.ndarray]]:
+    ) -> Union[float, list, tuple[float, np.ndarray], list[tuple[float, np.ndarray]]]:
         """
-        This method calls the forward model via problem.evaluate(inputs),
-        and computes the cost for the given output by calling self.compute().
+        Compute cost and optional gradient for given input parameters.
 
         Parameters
         ----------
-        inputs : Inputs or list-like
-            The input parameters for which the cost and optionally the gradient
-            will be computed.
-        calculate_grad : bool, optional, default=False
+        inputs : Union[Inputs, list, np.ndarray]
+            Input parameters for cost computation. Supports single or multiple inputs.
```

Comment:
needs more info, how do you do single or multiple inputs. Are the different inputs on the rows or columns of the ndarray?